refactor(linter): add missing `should_run` implementations

camchenry opened the pull request:
Added some `should_run` implementations that probably should exist but didn't previously.

camchenry commented:
* **#6666** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6666?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camchenry and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

DonIsaac commented on crates/oxc_linter/src/rules/typescript/no_namespace.rs:
```diff
@@ -104,6 +100,9 @@ impl Rule for NoNamespace {
     }
 
     fn should_run(&self, ctx: &ContextHost) -> bool {
```
Looks like a bug? We're returning false when definition files are allowed and the current file is a definition file?

camchenry replied:
I think it's correct: if we're looking at a - definition file and definition files are allowed, then we don't need to emit any diagnostics do should_run be false. Does that seem incorrect?

DonIsaac replied:
I think I misunderstood the meaning of "allow definition files". I believe you're correct

DonIsaac commented:
### Merge activity

* **Oct 19, 1:53 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
refactor(linter): add missing `should_run` implementations (#6666)

Added some `should_run` implementations that probably should exist but didn't previously.
